### DISCUSSION TITLE
Remix CC Q4 工作规划

### DISCUSSION THREAD

**Discussion by drafish** [PlanckerLabs/remix-cc#89] [Show and tell]
### Remix CC 官网
展示 Remix CC 的使命与愿景、在做哪些项目、团队成员、财务公示、赞助方、活动信息等等

**交付成果:** 
1. Remix CC 官网

### 镜像站维护
Remix IDE 镜像站、Remix 文档镜像站、Remix 静态资源镜像站

**交付成果**
1. 保证镜像站与 Remix 官方同步更新

### 非 EVM 兼容链的 Remix 插件模板
非 EVM 兼容链只能通过 iframe 插件集成到 Remix 上。Remix 上现有两个非 EVM 兼容链的插件，Celo 和 Starknet ，交互逻辑和UI样式都不统一。未来还可能会有更多的非 EVM 兼容链集成到 Remix 上。这些插件的功能其实都差不多，核心功能就是编译+部署调用。我觉得可以按 Remix 官方的功能为样本，为非 EVM 兼容链打造一个模板。这样做有两个好处
1. 可以降低各家非 EVM 兼容链接入 Remix 的成本
2. 统一交互逻辑和样式，降低用户的学习成本

**交付成果**
1. 非 EVM 兼容链的 Remix 插件模板

**备注**
remix 团队表示愿意捐赠，但需要我们写一个spec

### Remix IDE 国际化功能维护
随着 Remix IDE 后续的迭代升级，又会产生很多新的 UI 文案，需要有人把这些新增的文案转化成国际化组件。

**交付成果**
1. Remix IDE 国际化 PR

**备注**
remix 团队表示愿意捐赠，具体金额还没详谈

### Remix 文档翻译
翻译四季度新增文档，组织中文社区力量一起审阅文档

**交付成果**
1. crowdin 上的翻译记录
3. 组织协作翻译审阅的文档

### 教学视频
制作原创视频、翻译 Remix 官方的视频，片尾加上赞助方信息

**交付成果**
1. 6 个原创视频
3. 7 个翻译视频

**备注**
remix 团队表示愿意捐赠，原创视频暂定是按一天一个的工作量来评估，每个视频 8*22U

### twitter space 采访
接受普朗克和 GCC 联合采访

**交付成果**
1. 采访记录

### 星火链插件
为星火链开发一个 Remix 插件

**交付成果**
1. 星火链插件

### 新版 OneClickDapp
Remix 上原有的 OneClickDapp 已经停止维护了。Remix 官方正考虑后续怎么处理这个插件，详见
- https://github.com/ethereum/remix-project/issues/3353

FootPrint 有意向开发一个新版 OneClickDapp ，并且在插件中集成他们的 dapp 数据展示能力。
四季度 Remix CC 会和 FootPrint 联合开发新版 OneClickDapp 。各方角色如下
1. Remix CC 是产品经理
2. FootPrint 是开发
5. Remix 团队是顾问

**交付成果**
1. 新版 OneClickDapp 产品文档

**备注**
remix 团队表示愿意捐赠，并且可以和 remix 团队一起讨论

### 新版 Remix VSCode
四季度会与 Remix 团队讨论下关于新版 Remix VSCode 的想法，看看能得到什么支持。并且继续寻找合适的开发工程师加入。相关背景资料详见
- https://github.com/PlanckerLabs/remix-cc/discussions/65

**交付成果**
1. 链接 Remix 官方的 VSCode 插件的作者
3. 链接 Remix-Light VSCode 插件的作者
4. 链接 Solidity Debugger VSCode 插件的作者 
6. 链接中文区其他感兴趣的小伙伴，比如说 moxii

### Remix git 功能优化
目前 Remix 的 git 功能只支持 github 。我觉得可以支持的更多，比如 gitlab、bitbucket、gitee 等等，局域网内的 git 服务也可以支持一下。

**交付成果**
1. issue 讨论需求
2. PR 提交代码

**备注**
remix 团队表示愿意捐赠，具体金额还没详谈

### LearnETH 插件重构
learneth 插件是用 angular 开发的，原来的开发者已经离职了。目前 remix 团队的主技术栈用的是react，所以想用 react 重构这个插件，方便后期维护。另外，我还发现了一个 bug 和一个可优化的点，也可以顺道一块儿做了

**交付成果**
1. 重构后的代码

**备注**
remix 团队表示愿意捐赠，具体金额还没详谈

### 运营
四季度有 yiwen 同学加入负责 Remix CC 的运营工作。Remix CC 产出的文章和视频会覆盖到更多的平台。各方面运营数据应该会较三季度有大幅提升

**交付成果**
1. 各平台的运营数据

### workshop
翻译 Remix 官方的 workshop PPT，在中文社区组织一次 workshop。这个事儿 Remix 那边提了好几次了，他们挺想看到其他语种的社区也举办 Remix workshop。四季度安排上

**交付成果**
1. workshop 视频

**Comment by drafish** [comment]
### Q4 预计固定支出
1. DJQ 3500U*3
2. yiwen 2500U*3

### Q4 预计收入
1. 外部合作 1150 U
2. Remix 文档翻译 528U

### GCC 申请
希望能从 GCC 申请到 16322U

### 不确定收入
Q4 会和 Remix 沟通下扩大捐赠范围，但目前还不确定。

**Comment by yiwen4** [comment]
如果你对某个 **待认领** 的任务感兴趣，请先了解项目的详情确定可投入时间，希望获得多少报酬，达成一致后可移交任务。


|  任务名称   | 任务描述  | 认领人   |  报酬 |  希望你|  
|  ----  | ----  |  ----  | ----  | ----  |
| 非 EVM 兼容链的 Remix 插件模板  | 把remix中的solidity compiler、deploy & run、debugger模块抽离出来集成到一个iframe插件里，可复用到别的链。 https://github.com/PlanckerLabs/remix-compiler-plugin https://github.com/PlanckerLabs/remix-udapp-plugin | 待认领 | 详谈 |  对remix源码比较熟悉 |
|   教学视频制作6个  | 根据对Remix IDE的了解，制作教学视频。 https://space.bilibili.com/3546557343337225  | Mocca2个，还有4个待认领 | bounty |  对Remix IDE熟悉  |
|   新版 Remix VSCode  | 计划开发Remix VSCode版  https://github.com/PlanckerLabs/remix-cc/discussions/96 | 待认领 | 详谈 | 对Remix和VScode开发都熟悉  |
|   Remix IDE 国际化功能维护 | Remix IDE新增文案转化成国际化组件 | DJQ |  CBI  |    |
|  Remix CC 官网  | 展示 Remix CC 详细信息  |  yiwen | CBI |    |
|   镜像站维护  | 维护Remix IDE、文档、静态资源镜像站与 Remix 官方同步更新 | DJQ | CBI |    |
|   Remix 文档翻译  | 翻译四季度新增文档 | yiwen | CBI |   |
|   Twitter space 采访  | 普朗克 GCC联合采访 | DJQ |   |    |
|   制作翻译教学视频  | 翻译Remix官方教学视频7个 | yiwen |  |    |
|   星火链插件  | 为星火链开发一个 Remix 插件 |  DJQ | CBI  |    |
|   Remix git 功能优化  |   优化Remix 的 git 功能，使其支持gitlab、bitbucket、gitee | DJQ | CBI  |    |
|   LearnETH 插件维护 | 帮助Remix 官方维护这个插件 | DJQ | CBI  |    |
|   Remix CC全平台的运营 |  https://github.com/PlanckerLabs/remix-cc/discussions/93 | yiwen | CBI  |    
|   Workshop | 翻译 Remix 官方的 Workshop PPT，在中文社区组织一次 Workshop | DJQ   yiwen | CBI  |    |
